posthog/temporal/batch_exports/s3_batch_export.py

```diff
@@ -702,25 +753,24 @@ async def insert_into_s3_activity(inputs: S3InsertInputs) -> RecordsCompleted:
             [field.with_nullable(True) for field in record_batch_schema]
         )
 
-        async with s3_upload as s3_upload:
-            records_completed = await run_consumer_loop(
-                queue=queue,
-                consumer_cls=S3Consumer,
-                producer_task=producer_task,
-                heartbeater=heartbeater,
-                heartbeat_details=details,
-                data_interval_end=data_interval_end,
-                data_interval_start=data_interval_start,
-                schema=record_batch_schema,
-                writer_format=WriterFormat.from_str(inputs.file_format, "S3"),
-                max_bytes=settings.BATCH_EXPORT_S3_UPLOAD_CHUNK_SIZE_BYTES,
-                s3_upload=s3_upload,
-                include_inserted_at=True,
-                writer_file_kwargs={"compression": inputs.compression},
-            )
-
-            await s3_upload.complete()
-
+        records_completed = await run_consumer_loop(
+            queue=queue,
+            consumer_cls=S3Consumer,
+            producer_task=producer_task,
+            heartbeater=heartbeater,
+            heartbeat_details=details,
+            data_interval_end=data_interval_end,
+            data_interval_start=data_interval_start,
+            schema=record_batch_schema,
+            writer_format=WriterFormat.from_str(inputs.file_format, "S3"),
+            max_bytes=settings.BATCH_EXPORT_S3_UPLOAD_CHUNK_SIZE_BYTES,
+            s3_upload=s3_upload,
+            include_inserted_at=True,
+            max_file_size_bytes=settings.BATCH_EXPORT_S3_UPLOAD_MAX_FILE_SIZE_BYTES,
```

Comment:
I'm not fully understanding how the `max_file_size_bytes` and `max_bytes` settings interact together. What happens if we configure a `max_file_size_bytes` lower than `max_bytes`? Let's review this together as I'm getting confused by how the different combinations of `max_bytes` and `max_file_bytes` play out together.